src/app/server-cluster/ServerClusterInterface.h

```diff
@@ -0,0 +1,198 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/AttributeValueDecoder.h>
+#include <app/AttributeValueEncoder.h>
+#include <app/CommandHandler.h>
+#include <app/ConcreteClusterPath.h>
+#include <app/data-model-provider/ActionReturnStatus.h>
+#include <app/data-model-provider/MetadataList.h>
+#include <app/data-model-provider/MetadataTypes.h>
+#include <app/data-model-provider/OperationTypes.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/BitFlags.h>
+
+namespace chip {
+namespace app {
+
+/// Defines an active cluster on an endpoint.
+///
+/// Provides metadata as well as interaction processing (attribute read/write and command handling).
+class ServerClusterInterface
+{
+public:
+    ServerClusterInterface();
```

Comment:
I don't see any methods to associate the Matter stack context, that would give access to:

- EventLogging
- Attribute dirty marking
- PersistentStorageDelegate
- FabricTable
- SceneTable (if it exists)
- etc

Which is all state needed by several clusters. Is it intended that this would be done not by common code? Perhaps a few clusters don't need to log events, but I claim most would need to mark dirty.

Reply:
I will have to design how the common context looks like and is initialized. it very likely it is a datamodelprovider (provides a context with event logging, eventing, dirty notification) however will see what else. My concern is that i have to balance:
   - provide everything via a pointer - minimal RAM, however context will be 100% a god object that will be really hard for tests to provide
   - provide a few choices/reduced amount or defer to clusters themselves (so that main can set them up) but that uses more RAM and ember has a hard time.
   
 I am leaning towards followup to refine things.

Reply:
OK, we can follow-up. This could be partially solved by adding interface methods for those common cases of marking dirty or logging an event, that would then access the right state when it becomes registered. That way the API accounts for it today.

Reply:
Keeping it as followup for now, so we can have the starter PR in. It is reasonably large already.

Followup I thinkg would be:
  - iterators (seems straight-forward now to have listings of clusters and attributes based on the registry)
  - context design
  - have some clusters converted
  
  Would like to be able to see flash overhead as soon as possible, hence having things in sooner is better.